components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -530,6 +605,14 @@ void AIChatTabHelper::WebContentsDestroyed() {
       ->RemoveObserver(this);
 }
 
+data_decoder::DataDecoder* AIChatTabHelper::GetDataDecoder() {
+  if (!data_decoder_) {
+    VLOG(1) << "Creating DataDecoder for APIRequestHelper";
+    data_decoder_ = std::make_unique<data_decoder::DataDecoder>();
+  }
+  return data_decoder_.get();
+}
+
```

Comment:
As discussed over DM, it probably makes sense to remove this (`GetDataDecoder` and `data_decoder_`) and instead directly call `DataDecoder::ParseXmlIsolated` since this is an infrequent call (only happens once for a Page load when the SidePanel is opened and doesn't happen other times e.g. session restore).